plugins/catalog-backend/src/database/Database.ts

```diff
@@ -388,18 +393,65 @@ export class Database {
     }
   }
 
-  async location(id: string): Promise<DbLocationsRow> {
-    const items = await this.database<DbLocationsRow>('locations')
-      .where({ id })
-      .select();
+  async location(id: string): Promise<DbLocationsRowWithStatus> {
+    const items = await this.database<DbLocationsRowWithStatus>('locations')
+      .where('locations.id', id)
+      .leftJoin(
```

Comment:
should probably be an outer join, shouldn't it? unless we guarantee that an event is present the same instant that the location is inserted, an inner join will make it look like the location doesn't exist until a log event comes in

Reply:
- [x] `leftJoin` -> `leftOuterJoin`